README.md
# Sigma Credtis Client
Sigma Credits Client es una herramienta gráfica que facilita la interacción con la API de [Sigma Credits](https://sigma-search.io)

<p align="center">
  <img src="https://user-images.githubusercontent.com/74129955/185270717-b02838cf-d58b-4f37-a9f7-322263561e36.png" />
</p>

## Requerimientos
```console
aiohttp==3.7.4.post0
PyQt5==5.15.7
```

## Instalación
### Source
Se puede clonar el repositorio y correr el programa desde el script *app.py*
```console
git clone https://github.com/SigmaCorp/Sigma-Credits-Client
cd Sigma-Credits-Client
pip install -r requirements.txt
python app.py
```

## Creditos
Todos los derechos reservados por Sigma Corporation.


CHANGELOG.md
# Change Log
Todos los cambios realizados en Sigma Client estan listados a continuación.

## Version 0.7.1 (2022-10-13)
### Implementaciones
- Exportar datos en CSV

## Version 0.7.0 (2022-09-03)
### Implementaciones
- Acceso a busqueda en data breach para plan medium o superior

### Fixed
- Peru resolver reference bug

## Version 0.6.9 (2022-09-02)
### Changed
- Cambio en los endpoints a la version 2

## Version 0.6.8 (2022-08-30)
### Changed
- Cambio en los endpoints de resolucion de DNI para profesional

## Version 0.6.5 (2022-08-23)
### Implementaciones
- Incluido endpoints limitados para comunidades
- Incluido la posibilidad de enviar reportes a nuestro programa de bug bounty

### Changed
- La busqueda de nombres incluye busqueda por DNI

## Version 0.6.2 (2022-08-19)
### Implementaciones
- Filtros para la busqueda de nombres

### Changed
- Arreglos en la busqueda de nombres

## Version 0.6.0 (2022-08-18)
### Implementaciones
- El cliente avisa al usuario de nuevas actualizaciones
- Se muestra el changelog si existe una version mas actual
- Resolver para buscar titular de un email
- Resolver para buscar titular de CBU/CVU o alias

### Changed
- El buscador de DNI profesional ya no requiere genero de la persona

## Version 0.5.3 (2022-08-15)
### Implementaciones
- Resolver para DNI profesional con información de obras sociales
- Busqueda de personas por nombres, apellidos y CUIT
- Busqueda de vecinos por dirección
- Login por usuario y password

### Changed
- Ya no se utiliza token para loguearse en el cliente
- Los resolvers internacionales free requieren al menos una cuenta

application/sdk.py
import aiohttp
import json
from typing import Any, ClassVar, Dict


class Route:
    BASE: ClassVar[str] = "https://sigma-search.io/api/sigma"

    def __init__(self, method: str, path: str):
        self.method = method
        self.path = path

    @property
    def url(self) -> str:
        return f"{self.BASE}{self.path}"


class SigmaCreditSDK:
    def __init__(self, token: str):
        self.token = token

    async def request(self, route: Route, **kwargs: Any) -> Any:
        url = route.url
        method = route.method

        headers: Dict[str, str] = {
            "User-Agent": "Sigma Credits Client - Integrated SDK v1.0",
            "sigma-key": self.token,
        }

        if "json" in kwargs:
            headers["Content-Type"] = "application/json"
            kwargs["data"] = json.dumps(kwargs.pop("json"))

        kwargs["headers"] = headers

        async with aiohttp.ClientSession() as session:
            async with session.request(method, url, **kwargs) as response:
                if "application/json" in response.headers["content-type"]:
                    rcontent = await response.json()
                else:
                    rcontent = await response.text()

        return rcontent

    def balance(self):
        return self.request(
            Route("POST", "/creditos"), json={"token": self.token}
        )

    def search_dni(self, dni: str, gender: str):
        data = {
            "metodo": "buscar_dni",
            "args": {
                "dni": dni,
                "sexo": gender,
            },
        }
        return self.request(Route("POST", "/sigma_credits"), json=data)

    def search_plate(self, plate: str):
        data = {
            "metodo": "buscar_patentes",
            "args": {"patente": plate},
        }
        return self.request(Route("POST", "/sigma_credits"), json=data)

    def search_phone(self, phone: str):
        data = {
            "metodo": "buscar_celular",
            "args": {"numero": phone},
        }
        return self.request(Route("POST", "/sigma_credits"), json=data)

    def search_name(
        self,
        name: str,
        province: int = -1,
        city: str = "",
        minage: int = 0,
        maxage: int = 0,
    ):
        args = {"nombre": name}

        if province >= 0:
            args["provincia_id"] = province

        if city:
            args["localidad"] = city

        if minage:
            args["edad_desde"] = minage

        if maxage:
            args["edad_hasta"] = maxage

        data = {
            "metodo": "buscar_nombre",
            "args": args,
        }
        return self.request(Route("POST", "/sigma_credits"), json=data)

    def search_address(self, address: str):
        data = {
            "metodo": "buscar_direccion",
            "args": {"direccion": address},
        }
        return self.request(Route("POST", "/sigma_credits"), json=data)

    def search_email(self, email: str):
        data = {
            "metodo": "buscar_datos",
            "args": {"tipo": "buscar_email", "dato": email},
        }
        return self.request(Route("POST", "/sigma_credits"), json=data)

    def search_number(self, number: str):
        data = {
            "metodo": "buscar_datos",
            "args": {"tipo": "buscar_celular", "dato": number},
        }
        return self.request(Route("POST", "/sigma_credits"), json=data)

    def search_cbu(self, cbu_alias: str):
        data = {
            "metodo": "buscar_datos",
            "args": {"tipo": "buscar_cbu_alias", "dato": cbu_alias},
        }
        return self.request(Route("POST", "/sigma_credits"), json=data)

    def search_breach(self, query: str):
        data = {
            "metodo": "buscar_breach",
            "args": {"query": query},
        }
        return self.request(Route("POST", "/sigma_credits"), json=data)

    @classmethod
    async def from_login(cls, username: str, password: str):
        sdk = cls("")
        res = await sdk.request(
            Route("POST", "/client/login"),
            json={"username": username, "password": password},
        )
        sdk.token = res["token"]
        return sdk


async def testing():
    sclient = await SigmaCreditSDK.from_login("notcia", "barto1337@!L")
    results = await sclient.balance()
    print(results)


if __name__ == "__main__":
    import asyncio

    loop = asyncio.get_event_loop()
    loop.run_until_complete(testing())


application/client.py
import asyncio

from datetime import datetime
from PyQt5 import QtWidgets, QtGui, QtCore

from .interface import Ui_MainWindow
from .sdk import SigmaCreditSDK
from .utils import retrieve_data
from .entries import PersonaEntry

__version__ = "0.1.0"


class SigmaClient(Ui_MainWindow):
    def __init__(self, mainwindow):
        super().__init__()
        self.mainwindow = mainwindow
        self.setupUi(mainwindow)
        self.__connect_signals()
        self.__bloquear_peticiones()
        self.sigma_version_label.setText(f"Sigma Credits v{__version__}")
        self.mostrar_mensaje(
            self.mconsole_textBrowser,
            f"Iniciando Sigma Credits build v{__version__}",
        )

        self.sdk = None
        self.local_credits = 0.00
        self.personas_cargadas = []

    def __connect_signals(self):
        self.validarLogin_pushButton.clicked.connect(self.validar_login)
        self.buscarPatente_pushButton.clicked.connect(self.buscar_por_patente)
        self.proBuscarNum_pushButton.clicked.connect(
            self.buscar_datos_por_numero
        )
        self.proBuscarDni_pushButton.clicked.connect(
            self.buscar_datos_profesional
        )

        self.proBuscarNombre_pushButton.clicked.connect(
            self.buscar_personas_nombre
        )

        self.proBuscarVecinos_pushButton.clicked.connect(
            self.buscar_vecinos_direccion
        )

        self.p3buscarcvuTitular_pushButton.clicked.connect(
            self.buscar_datos_cbu
        )

        self.p3emailBuscar_pushButton.clicked.connect(self.buscar_datos_email)

        self.p3buscarDato_pushButton.clicked.connect(
            self.buscar_datos_numero_p3
        )

        self.buscarLeaks_pushButton.clicked.connect(self.buscar_data_breach)

    def __bloquear_peticiones(self):
        self.buscarPatente_pushButton.setEnabled(False)
        self.proBuscarNum_pushButton.setEnabled(False)
        self.proBuscarDni_pushButton.setEnabled(False)
        self.proBuscarNombre_pushButton.setEnabled(False)
        self.proBuscarVecinos_pushButton.setEnabled(False)
        self.p3buscarcvuTitular_pushButton.setEnabled(False)
        self.p3emailBuscar_pushButton.setEnabled(False)
        self.p3buscarDato_pushButton.setEnabled(False)
        self.buscarLeaks_pushButton.setEnabled(False)

    def __habilitar_peticiones(self):
        self.buscarPatente_pushButton.setEnabled(True)
        self.proBuscarNum_pushButton.setEnabled(True)
        self.proBuscarDni_pushButton.setEnabled(True)
        self.proBuscarNombre_pushButton.setEnabled(True)
        self.proBuscarVecinos_pushButton.setEnabled(True)
        self.p3buscarcvuTitular_pushButton.setEnabled(True)
        self.p3emailBuscar_pushButton.setEnabled(True)
        self.p3buscarDato_pushButton.setEnabled(True)
        self.buscarLeaks_pushButton.setEnabled(True)

    def __limpiar_tabla(self, tabla):
        while tabla.rowCount() > 0:
            tabla.removeRow(0)

    def mostrar_mensaje(self, consola, msg: str):
        consola.append(f"[{datetime.now().strftime('%H:%M:%S')}] {msg}")

    def update_balance(self, balance: float):
        self.local_credits = balance
        self.MainWindow.setWindowTitle(
            f"Sigma Credits (Balance: ${self.local_credits})"
        )

    async def async_validar_login(self):
        usuario = self.usuario_lineEdit.text().strip()
        clave = self.password_lineEdit.text().strip()
        self.mostrar_mensaje(
            self.mconsole_textBrowser,
            f"Validando credenciales {usuario}:{clave} ...",
        )

        try:
            sdk = await SigmaCreditSDK.from_login(usuario, clave)
        except:
            self.mostrar_mensaje(
                self.mconsole_textBrowser,
                "Las credenciales son invalidas o no es posible logearse en este momento",
            )
            return

        balance_response = await sdk.balance()
        balance = balance_response["creditos"]

        self.mostrar_mensaje(
            self.mconsole_textBrowser,
            f"Bienvenido {usuario}! Dispones de {balance} creditos",
        )
        self.update_balance(balance)
        self.sdk = sdk
        self.mostrar_mensaje(
            self.mconsole_textBrowser,
            f"Tu cliente ha sido configurado para utilizar la token {sdk.token}",
        )
        self.__habilitar_peticiones()

    async def async_buscar_por_patente(self):
        patente_str = self.patenteBuscar_lineEdit.text().strip()
        self.mostrar_mensaje(
            self.patentes_textBrowser,
            f"Buscando registros de la patente {patente_str} ...",
        )

        response = await self.sdk.search_plate(patente_str)

        if "error" in response:
            self.mostrar_mensaje(
                self.patentes_textBrowser, response.get("error")
            )
        else:
            self.mostrar_mensaje(
                self.patentes_textBrowser,
                f"Se encontraron {len(response)} registros relacionados a la patente {patente_str}, cargando ...",
            )
            self.update_balance(self.local_credits - 0.25)
            for registro in response:
                rowPosition = self.patentes_tableWidget.rowCount()
                self.patentes_tableWidget.insertRow(rowPosition)
                for indice, campo in enumerate(
                    [
                        "patente",
                        "documento",
                        "vehiculo",
                        "marca",
                        "anio",
                        "titular",
                        "porcentaje",
                        "calle",
                        "altura",
                        "piso",
                        "depto",
                        "codigo_postal",
                        "localidad",
                        "transferencia",
                    ]
                ):
                    self.patentes_tableWidget.setItem(
                        rowPosition,
                        indice,
                        QtWidgets.QTableWidgetItem(
                            retrieve_data(registro, campo)
                        ),
                    )

    async def async_buscar_datos_por_numero(self):
        numero_str = self.proNumeroBuscar_lineEdit.text().strip()
        self.mostrar_mensaje(
            self.numeros_textBrowser,
            f"Buscando datos relacionados al numero {numero_str} ...",
        )

        response = await self.sdk.search_phone(numero_str)

        if "error" in response:
            self.mostrar_mensaje(
                self.numeros_textBrowser, response.get("error")
            )
        else:
            self.mostrar_mensaje(
                self.numeros_textBrowser,
                f"Se encontraron {len(response)} entidades relacionadas al numero {numero_str}, cargando datos ...",
            )
            self.update_balance(self.local_credits - 0.25)
            for entidad in response:
                rowPosition = self.proNumeros_tableWidget.rowCount()
                self.proNumeros_tableWidget.insertRow(rowPosition)
                for indice, campo in enumerate(
                    [
                        "celular",
                        "documento",
                        "nombre",
                        "direccion",
                        "localidad",
                        "provincia",
                        "codigo_postal",
                        "empresa",
                    ]
                ):
                    self.proNumeros_tableWidget.setItem(
                        rowPosition,
                        indice,
                        QtWidgets.QTableWidgetItem(
                            retrieve_data(entidad, campo)
                        ),
                    )

    async def async_buscar_datos_profesional(self):
        dni_str = self.proDniBuscar_lineEdit.text().strip()
        genero_str = self.proGenero_comboBox.currentText()
        self.mostrar_mensaje(
            self.personas_textBrowser,
            f"Buscando datos relacionados al dni {dni_str} ...",
        )

        response = await self.sdk.search_dni(dni_str, genero_str)

        if "error" in response:
            self.mostrar_mensaje(
                self.personas_textBrowser, response.get("mensaje")
            )
        else:
            self.mostrar_mensaje(
                self.personas_textBrowser,
                f"Se ha encontrado el DNI {dni_str}, cargando datos ...",
            )
            self.update_balance(self.local_credits - 0.25)
            self.proEmision_output_lineEdit.setText(
                retrieve_data(response, "emision")
            )
            self.proNroDoc_output_lineEdit.setText(
                retrieve_data(response, "documento")
            )
            self.proApellido_output_lineEdit.setText(
                retrieve_data(response, "apellido")
            )
            self.proNombre_output_lineEdit.setText(
                retrieve_data(response, "nombres")
            )
            self.proCUIL_output_lineEdit.setText(
                retrieve_data(response, "cuil")
            )
            self.proBarrio_output_lineEdit.setText(
                retrieve_data(response, "barrio")
            )
            self.proCalle_output_lineEdit.setText(
                retrieve_data(response, "calle")
            )
            self.proAltura_output_lineEdit.setText(
                retrieve_data(response, "numero")
            )
            self.proDepartamento_output_lineEdit.setText(
                retrieve_data(response, "departamento")
            )
            self.proPiso_output_lineEdit.setText(
                retrieve_data(response, "piso")
            )
            self.proMonoblock_output_lineEdit.setText(
                retrieve_data(response, "monoblock")
            )
            self.proCiudad_output_lineEdit.setText(
                retrieve_data(response, "ciudad")
            )
            self.proMunicipio_output_lineEdit.setText(
                retrieve_data(response, "municipio")
            )
            self.proProvincia_output_lineEdit.setText(
                retrieve_data(response, "provincia")
            )
            self.proPais_output_lineEdit.setText(
                retrieve_data(response, "pais")
            )

            self.proFechaNacimiento_output_lineEdit.setText(
                retrieve_data(response, "fecha_nacimiento")
            )

            fallecimiento = retrieve_data(response, "fallecido")
            if fallecimiento == "Sin Aviso de Fallecimiento":
                fallecimiento = "No"

            self.proFallecido_output_lineEdit.setText(fallecimiento)

            if type(response.get("cobertura")) == list:
                for obrasocial in response.get("cobertura"):
                    self.obraSocial_output_comboBox.addItem(
                        obrasocial["cobertura"]
                    )
            else:
                self.obraSocial_output_comboBox.clear()

    async def async_buscar_personas_nombre(self):
        nombre_str = self.proNombreBuscar_lineEdit.text().strip()
        provincia_str = self.proProvincia_comboBox.currentText() or None
        localidad_str = self.proLocalidadBuscar_lineEdit.text().strip() or None
        edad_min_str = self.proEdadMin_spinBox.value() or None
        edad_max_str = self.proEdadMax_spinBox.value() or None

        self.mostrar_mensaje(
            self.personas_textBrowser,
            f"Buscando personas con nombre '{nombre_str}' ...",
        )

        response = await self.sdk.search_name(
            nombre_str,
            {
                None: -1,
                "capital_federal": 0,
                "buenos_aires": 1,
                "catamarca": 2,
                "chaco": 16,
                "chubut": 17,
                "cordoba": 3,
                "corrientes": 4,
                "entre_rios": 5,
                "formosa": 18,
                "jujuy": 6,
                "la_pampa": 21,
                "la_rioja": 8,
                "mendoza": 7,
                "misiones": 19,
                "neuquen": 20,
                "rio_negro": 22,
                "salta": 9,
                "san_juan": 10,
                "san_luis": 11,
                "santa_cruz": 23,
                "santa_fe": 12,
                "santiago_del_estero": 13,
                "tierra_del_fuego": 24,
                "tucuman": 14,
            }[provincia_str],
            localidad_str,
            edad_min_str,
            edad_max_str,
        )

        if "error" in response:
            self.mostrar_mensaje(
                self.personas_textBrowser, response.get("error")
            )
        else:
            self.mostrar_mensaje(
                self.personas_textBrowser,
                f"Se encontraron {len(response)} resultados, cargando ...",
            )
            self.update_balance(self.local_credits - 0.25)
            for persona in response:

                entry = PersonaEntry(
                    self.scrollArea,
                    persona["documento"],
                    persona["nombre"],
                    persona["provincia"],
                )
                self.verticalLayout_2.addWidget(entry)
                self.scrollArea.setWidget(self.scrollAreaWidgetContents)
                self.verticalLayout.addWidget(self.scrollArea)
                self.personas_cargadas.append(entry)

    async def async_buscar_vecinos_direccion(self):
        direccion_str = self.proDireccionBuscar_lineEdit.text().strip()
        self.mostrar_mensaje(
            self.personas_textBrowser,
            f"Buscando vecinos en {direccion_str} ...",
        )

        response = await self.sdk.search_address(direccion_str)

        if "error" in response:
            self.mostrar_mensaje(
                self.personas_textBrowser, response.get("error")
            )
        else:
            self.mostrar_mensaje(
                self.personas_textBrowser,
                f"Se encontraron {len(response)} vecinos en {direccion_str}, cargando datos ...",
            )
            self.update_balance(self.local_credits - 0.25)
            for vecino in response:
                rowPosition = self.proVecinos_tableWidget.rowCount()
                self.proVecinos_tableWidget.insertRow(rowPosition)
                for indice, campo in enumerate(
                    [
                        "doc",
                        "nombre",
                        "numero",
                        "empresa",
                        "direccion",
                        "localidad",
                        "provincia",
                        "codigo_postal",
                    ]
                ):
                    self.proVecinos_tableWidget.setItem(
                        rowPosition,
                        indice,
                        QtWidgets.QTableWidgetItem(
                            retrieve_data(vecino, campo)
                        ),
                    )

    async def async_buscar_datos_cbu(self):
        cbv_or_alias_str = self.p3cbvubuscar_lineEdit.text().strip()
        self.mostrar_mensaje(
            self.banking_textBrowser,
            f"Buscando datos del titular de {cbv_or_alias_str} ...",
        )

        response = await self.sdk.search_cbu(cbv_or_alias_str)

        if "error" in response:
            self.mostrar_mensaje(
                self.banking_textBrowser, response.get("mensaje")
            )
        else:
            self.mostrar_mensaje(
                self.banking_textBrowser,
                f"Se encontraron datos del titular de {cbv_or_alias_str}, cargando ...",
            )
            self.update_balance(self.local_credits - 0.25)
            self.p3titular_output_lineEdit.setText(
                retrieve_data(response, "nombre")
            )
            self.p3cuit_output_lineEdit.setText(
                retrieve_data(response, "cuit")
            )
            self.p3banco_output_lineEdit.setText(
                retrieve_data(response, "banco")
            )
            self.p3tipocuenta_output_lineEdit.setText(
                retrieve_data(response, "cuenta_tipo")
            )
            self.p3cbucvu_output_lineEdit.setText(
                retrieve_data(response, "cbu")
            )

    async def async_buscar_datos_email(self):
        email_str = self.p3emailBuscar_lineEdit.text().strip()
        self.mostrar_mensaje(
            self.banking_textBrowser,
            f"Buscando datos de email {email_str} ...",
        )

        response = await self.sdk.search_email(email_str)

        if "error" in response:
            self.mostrar_mensaje(
                self.banking_textBrowser, response.get("error")
            )
        else:
            self.mostrar_mensaje(
                self.banking_textBrowser,
                f"Se encontraron datos del email {email_str}, cargando ...",
            )
            self.update_balance(self.local_credits - 0.25)
            self.p3nombres_output_lineEdit.setText(
                retrieve_data(response, "nombre")
            )
            self.p3apellido_output_lineEdit.setText(
                retrieve_data(response, "apellido")
            )

    async def async_buscar_datos_numero_p3(self):
        numero_str = self.p3numeroBuscar_lineEdit.text().strip()
        self.mostrar_mensaje(
            self.numeros_textBrowser,
            f"Buscando datos del numero {numero_str} ...",
        )

        response = await self.sdk.search_number(numero_str)

        if "error" in response:
            self.mostrar_mensaje(
                self.numeros_textBrowser, response.get("error")
            )
        else:
            self.mostrar_mensaje(
                self.numeros_textBrowser,
                f"Se encontraron datos del numero {numero_str}, cargando ...",
            )
            self.update_balance(self.local_credits - 0.25)
            self.p3nombres2_output_lineEdit.setText(
                retrieve_data(response, "nombre")
            )
            self.p3apellidos2_output_lineEdit.setText(
                retrieve_data(response, "apellido")
            )
            self.p3email_output_lineEdit.setText(
                retrieve_data(response, "email")
            )
            self.p3numero_output_lineEdit.setText(
                retrieve_data(response, "numero")
            )

    async def async_buscar_data_breach(self):
        query_str = self.queryBreach_lineEdit.text().strip()

        self.mostrar_mensaje(
            self.banking_textBrowser,
            f"Buscando query {query_str} en nuestra DB ...",
        )

        response = await self.sdk.search_breach(query_str)

        if "error" in response:
            self.mostrar_mensaje(
                self.banking_textBrowser, response.get("error")
            )
        else:
            self.mostrar_mensaje(
                self.banking_textBrowser,
                f"Se encontraron {len(response)} entradas relacionados a la query {query_str}, cargando ...",
            )
            self.update_balance(self.local_credits - 0.25)
            for entrada in response:
                rowPosition = self.dataBreachResults_tableWidget.rowCount()
                self.dataBreachResults_tableWidget.insertRow(rowPosition)
                for indice, campo in enumerate(
                    [
                        "usuario",
                        "password",
                    ]
                ):
                    self.dataBreachResults_tableWidget.setItem(
                        rowPosition,
                        indice,
                        QtWidgets.QTableWidgetItem(
                            retrieve_data(entrada, campo)
                        ),
                    )

    def validar_login(self):
        self.__bloquear_peticiones()
        if (
            self.usuario_lineEdit.text().strip()
            and self.password_lineEdit.text().strip()
        ):
            asyncio.ensure_future(self.async_validar_login())

    def buscar_por_patente(self):
        self.__limpiar_tabla(self.patentes_tableWidget)
        if self.patenteBuscar_lineEdit.text().strip():
            asyncio.ensure_future(self.async_buscar_por_patente())

    def buscar_datos_por_numero(self):
        self.__limpiar_tabla(self.proNumeros_tableWidget)
        if self.proNumeroBuscar_lineEdit.text().strip():
            asyncio.ensure_future(self.async_buscar_datos_por_numero())

    def buscar_datos_profesional(self):
        if self.proDniBuscar_lineEdit.text().strip():
            asyncio.ensure_future(self.async_buscar_datos_profesional())

    def buscar_personas_nombre(self):
        if self.personas_cargadas:
            while self.verticalLayout_2.count():
                child = self.verticalLayout_2.takeAt(0)
                if child.widget() is not None:
                    child.widget().deleteLater()
            self.personas_cargadas = []
        if self.proNombreBuscar_lineEdit.text().strip():
            asyncio.ensure_future(self.async_buscar_personas_nombre())

    def buscar_vecinos_direccion(self):
        self.__limpiar_tabla(self.proVecinos_tableWidget)
        if self.proDireccionBuscar_lineEdit.text().strip():
            asyncio.ensure_future(self.async_buscar_vecinos_direccion())

    def buscar_datos_cbu(self):
        if self.p3cbvubuscar_lineEdit.text().strip():
            asyncio.ensure_future(self.async_buscar_datos_cbu())

    def buscar_datos_email(self):
        if self.p3emailBuscar_lineEdit.text().strip():
            asyncio.ensure_future(self.async_buscar_datos_email())

    def buscar_datos_numero_p3(self):
        if self.p3numeroBuscar_lineEdit.text().strip():
            asyncio.ensure_future(self.async_buscar_datos_numero_p3())

    def buscar_data_breach(self):
        self.__limpiar_tabla(self.dataBreachResults_tableWidget)
        if self.queryBreach_lineEdit.text().strip():
            asyncio.ensure_future(self.async_buscar_data_breach())
